### PR TITLE
Add error checking/logging for blank stock quantities

### PR DESCRIPTION
**Pull request by twymer**
http://manage.dimagi.com/default.asp?155093

If you submit xml with `quantity=""` it would just 500 and prevent you from syncing that form. This is an app problem, but we shouldn't break over it.

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/commtrack/models.py [hunk 1]
```diff
@@ -503,20 +504,33 @@ def _yield_txns(section_id, quantity):
...
             values = [child for child in product_node]
             for value in values:
                 section_id = value.attrib.get('section-id')
-                quantity = float(value.attrib.get('quantity'))
-                for txn in _yield_txns(section_id, quantity):
-                    yield txn
+                quantity = value.attrib.get('quantity')
+                # make sure quantity is not an empty, unset node value
+                if str(quantity).strip() != '':
+                    for txn in _yield_txns(section_id, flaot(quantity)):
```
Comment (millerdev) [inline corehq/apps/commtrack/models.py]: Should that be `float`?

File: corehq/apps/commtrack/models.py [hunk 2]
```diff
@@ -503,20 +504,33 @@ def _yield_txns(section_id, quantity):
...
         section_id = action_node.attrib.get('section-id', None)
         grouped_entries = section_id is not None
         if grouped_entries:
-            quantity = float(product_node.attrib.get('quantity'))
-            for txn in _yield_txns(section_id, quantity):
-                yield txn
-
+            quantity = product_node.attrib.get('quantity')
+            # make sure quantity is not an empty, unset node value
+            if str(quantity).strip() != '':
```
Comment (millerdev) [inline corehq/apps/commtrack/models.py]: Would it make sense to check for other non-float values as well? Something like this:
```python
def float_or_none(value):
    try:
        return float(value)
    except (ValueError, TypeError):
        log.error("Blank transaction quantity submitted on domain %s for "
                  "a %s ledger", config.domain, section_id)
        return None

...

quantity = float_or_none(product_node.attrib.get('quantity'))
if quantity is not None:
    for txn in ...
```

File: corehq/apps/commtrack/models.py (hunk 1 above)
Comment (twymer) [inline corehq/apps/commtrack/models.py]: .... Wow. I thought both paths were hit by the test, will add better coverage lol

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by twymer** [comment]
@millerdev Totally ripped off your idea :tada:

**Pull request merged**